features/vaadin-dashlets/dashlet-ksc/src/main/java/org/opennms/features/vaadin/dashboard/dashlets/KscDashlet.java

```diff
@@ -405,23 +407,16 @@ public Component getComponent() {
             public Map<String, String> doInTransaction(TransactionStatus transactionStatus) {
                 Map<String, String> data = new HashMap<String, String>();
 
-                if (nodeId == null) {
-                    String[] arr0 = resourceId.split("\\.");
-                    String[] arr1 = arr0[0].split("[\\[\\]]");
-                    data.put("nodeId", arr1[1]);
-                } else {
-                    data.put("nodeId", nodeId);
-                }
-
-                data.put("nodeLabel", m_nodeDao.getLabelForId(Integer.valueOf(data.get("nodeId"))));
+                String nodeIdentifier = Optional.ofNullable(nodeId).orElse(determineNodeIdByResourceId(resourceId));
+                data.put("nodeId", nodeIdentifier);
+                data.put("nodeLabel", m_nodeDao.getLabelForId(Integer.valueOf(nodeIdentifier)));
 
-                List<OnmsResource> resourceList = m_resourceDao.getResourceById(ResourceId.get("node", data.get("nodeId"))).getChildResources();
+                List<OnmsResource> resourceList = m_resourceDao.getResourceById(ResourceId.get("node", nodeIdentifier)).getChildResources();
 
                 for (OnmsResource onmsResource : resourceList) {
                     if (resourceId.equals(onmsResource.getId())) {
```

Comment:
I also found this problem which is unrelated to the NMS-10191:
This should never be true (and therefor we should either fix or remove this piece of code) since we compare her 2 different Types: String and ResourceId

Reply:
Exactly, I think this is broken since NMS-7548 where the ResourceId class was introduced. Before the resourceId was a simple String instance. Sure, this must be fixed.

Reply:
ok Ill fix it as well. Thanks!